### PR TITLE
fix: fix isOptedOut persistence, reset campaigns queue on user change, fix tests (SDKCF-3656)

### PR DESCRIPTION
**Pull request by imaginaris**
# Description

- Fix for unit tests that didn't detect the issue with opt-out
- isOptedOut flag is persisted now
- improvements for user preference update logic

## Links
SDKCF-3656

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: RInAppMessaging/Classes/CampaignDispatcher.swift [hunk 1]
```diff
@@ -15,17 +15,18 @@ internal protocol CampaignDispatcherType {
...
     private let permissionService: DisplayPermissionServiceType
     private let campaignRepository: CampaignRepositoryType
 
     private let dispatchQueue = DispatchQueue(label: "IAM.Campaign", qos: .userInteractive)
     private var queuedCampaigns = [Campaign]()
-    private(set) var isDispatching = false
+    @AtomicGetSet private(set) var isDispatching = false
 
     weak var delegate: CampaignDispatcherDelegate?
+    var scheduledTask: DispatchWorkItem?
```
Comment (donnie-jp) [inline RInAppMessaging/Classes/CampaignDispatcher.swift:29]: Where does this get set?
I only see a call to cancel()

Comment (donnie-jp) [reply RInAppMessaging/Classes/CampaignDispatcher.swift:29]: Ah, it's from TaskSchedulable

**Review by donnie-jp — APPROVED** [review]
LGTM

File: RInAppMessaging/Classes/CampaignDispatcher.swift (hunk 1 above)
Comment (imaginaris) [reply RInAppMessaging/Classes/CampaignDispatcher.swift:29]: That's right

**Pull request merged**